app/classes/Transvision/API.php

```diff
@@ -302,11 +323,17 @@ private function verifyEnoughParameters($number)
      * Check that the repository asked for is one we support
      *
      * @param  string  $repository Name of the repository
+     * @param  boolean $alias      Do we allow aliases for repository names,
+     *                             ex: 'global', to query all repositories. Default to False
      * @return boolean True if we support this repository, False if we don't
      */
-    private function verifyRepositoryExists($repository)
+    private function verifyRepositoryExists($repository, $alias = false)
     {
-        if (! in_array($repository, Project::getRepositories())) {
+        if (! in_array($repository, $this->valid_repositories)) {
```

Comment:
I think I lost your train of thoughts here. 

In the costructor you set `$this->valid_repositories` to all supported repositories plus `global`. So, `verifyRepositoryExists('global')` will return true, while it should return false, given that `global`is not an actual repository (e.g. `/api/v1/locales/global/` or `/api/v1/entity/global/?id=test`).

Also, you're not actually doing anything with the updated value of `$this->valid_repositories` after adding `global` if we accept an alias.

This method also seems to be the only user for `$this->valid_repositories`. Should we keep it as a class property?

I would expect something like this, without setting the value in the constructor
```
    private function verifyRepositoryExists($repository, $alias = false)
    {
        $this->valid_repositories = $alias
            ? array_merge(Project::getRepositories(), ['global'])
            : Project::getRepositories();
        if (! in_array($repository, $this->valid_repositories)) {
            $this->log("The repo queried ({$repository}) doesn't exist.");

            return false;
        }

        return true;
    }
```